### ISSUE TITLE
Add to Store: Duplicate Well getting created

### ISSUE THREAD

**Issue by bbuckles** [hashmapinc/Drillflow#216]
Issue: I can add store that already exists

PreCondition: Well UID already exists

Reproduction Steps:

1. Create Soap request for Add to Store for well UID that already exists

2. Send in Requests

Expected Results:

Error is thrown

Observed Behavior:

Receiving successful response (1) back and when I do a 'Get From Store' the Well UID was overwritten.

**Comment by randypitcherii** [comment]
Good catch. We have a `TODO` in the code to address this, but we didn't make a card yet.

We're currently not querying for a UID before sending a create request. To address this, we should first query DoT for an existing UID at the object type endpoint. If it's there, return an error to the client. If not, proceed.

Just for the record, it is absolutely insane to me that DoT does not throw an error for this behavior. It is an incredibly bad idea for it to expect individual rest clients to all know what is and is not acceptable behavior.

**Comment by cherrera2001** [comment]
Blocked by slb as they have to implement this logic.

**Comment by Gimli62** [comment]
> Lot of insane-ness as you have mentioned.

If you do believe that DoT APIs needs enhancement to remove what you qualify as "insane-ness" please make sure to discuss these points during weekly so that the DoT API is updated or discussed to improve the overall experience on both side .

Thanks
Pascal.

**Comment by cherrera2001** [comment]
@Gimli62 I think the issue here is one that was raised in one of our meetings, but not actually noted (our bad...). The problem is as all the posts are Upsert semantics, we have to do a get before we do a post. Is this something that we could just choose that all updates/adds are respected the same, or should we do a get before the post?

**Comment by niaalex** [comment]
SLB team has not yet responded.  HM team will continue to monitor this in BL.

**Comment by Mike-d-s** [comment]
Leaving this card in the well backlog in order to track that the UPSERT issue still needs to be resolved within the SLB API or on the DrillFlow side.
